module/zfs/zio.c

```diff
@@ -2856,6 +2856,20 @@ zio_nop_write(zio_t *zio)
 		ASSERT(bcmp(&bp->blk_prop, &bp_orig->blk_prop,
 		    sizeof (uint64_t)) == 0);
 
+		/*
+		 * If we're overwriting a block that is currently on an
+		 * indirect vdev, then ignore the nopwrite request and
+		 * allow a new block to be allocated on a concrete vdev.
+		 */
+		spa_config_enter(zio->io_spa, SCL_VDEV, FTAG, RW_READER);
+		vdev_t *tvd = vdev_lookup_top(zio->io_spa,
+		    DVA_GET_VDEV(&bp->blk_dva[0]));
+		if (tvd->vdev_ops == &vdev_indirect_ops) {
```

Comment:
if `tvd == NULL`, we will be failed in access to content. more safely to do the same like at:
https://github.com/zfsonlinux/zfs/blob/master/module/zfs/metaslab.c#L3791

Reply:
In this context we're looking at a blkptr that should be valid so the top-level vdev needs to still exist in the vdev namespace. If that tvd doesn't exist then we have bigger problems and should probably panic.